components/application-mgt/org.wso2.carbon.device.application.mgt.core/src/main/java/org/wso2/carbon/device/application/mgt/core/impl/CommentsManagerImpl.java

```diff
@@ -17,10 +17,274 @@
+        } finally {
+            ConnectionManagerUtil.closeDBConnection();
+        }
+        return null;
+    }
+
+    @Override
+    public Comment getComment(int apAppCommentId) throws CommentManagementException {
+
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId(true);
+        Comment comment=null;
+
+        if (log.isDebugEnabled()) {
+            log.debug("Comment retrieval request is received for the comment id " + apAppCommentId );
+        }
+
+        try {
+            ConnectionManagerUtil.openDBConnection();
+            comment=ApplicationManagementDAOFactory.getCommentDAO().getComment(apAppCommentId);
+        } catch (DBConnectionException e) {
+            log.error("DB Connection Exception occurs.", e);
+        } catch (SQLException e) {
+            log.error("SQL Exception occurs.", e);
+        } finally {
+            ConnectionManagerUtil.closeDBConnection();
+        }
+        return comment;
+    }
+
+    @Override
+    public void deleteComment(int apAppCommentId) throws CommentManagementException {
+
+        Comment comment;
+        comment= getComment(apAppCommentId);
+
+        if (comment == null) {
+            throw new CommentManagementException(
+                        "Cannot delete a non-existing comment for the application with comment id" + apAppCommentId);
+        }
+        try {
+            ConnectionManagerUtil.beginDBTransaction();
+            ApplicationManagementDAOFactory.getCommentDAO().deleteComment(apAppCommentId);
+            ConnectionManagerUtil.commitDBTransaction();
+        } catch (DBConnectionException e) {
+            log.error("DB Connection Exception occurs.", e);
+        } catch (SQLException e) {
+            log.error("SQL Exception occurs.", e);
+        } catch (TransactionManagementException e) {
+            log.error("Transaction Management Exception occurs.", e);
+        } finally {
+            ConnectionManagerUtil.closeDBConnection();
+        }
+    }
+
+    @Override
+    public Comment updateComment(Comment comment,int apAppCommentId) throws CommentManagementException, SQLException,
+            DBConnectionException {
+
+        PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantId(true);
+        validateComment(apAppCommentId,comment.getComment());
+
+        if (log.isDebugEnabled()) {
+            log.debug("Comment retrieval request is received for the comment id " +apAppCommentId);
+        }
+        comment.setModifiedAt(Timestamp.from(Instant.now()));
+        try {
+            ConnectionManagerUtil.openDBConnection();
+            ApplicationManagementDAOFactory.getCommentDAO().getComment(apAppCommentId);
+            return ApplicationManagementDAOFactory.getCommentDAO().updateComment(apAppCommentId,
+                    comment.getComment(),comment.getModifiedBy(),comment.getModifiedAt());
+        } catch (SQLException e) {
+            log.error("SQL Exception occurs.", e);
```

Comment:
if we throw `SQLException` this is not necessary. However; best practice is to wrap & throw this sql exception with `CommentManagementException` since throwing `SQLException`,`DBConnectionException ` from a manager class is not encouraged.